lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -57,80 +57,68 @@ def execute(self) -> int:
 
-                    tool_info[group][host]["tools"] = {}
-
-                    for tool in sorted(self.tools(path)):
-                        v = sorted((path / tool).read_text().rstrip("\n").split("\n"))
-                        tool_info[group][host]["tools"][tool] = v if opts else ""
-
-            if tool_info:
-                found = self.print_results(tool_info, self.context.with_option)
-                if not found:
-                    msg = "No tools found"
-                    if self.context.group:
-                        msg += f' in group "{self.context.group[0]}"'
-                    self.logger.warn(msg)
-            else:
-                self.logger.warn("No tool groups found")
-
-            return 0
-
-        else:
-            # List the groups which include this tool
-            tool = self.context.name
-            found = False
-            for group in groups:
-                try:
-                    tg_dir = self.gen_tools_group_dir(group)
-                except BadToolGroup:
-                    self.logger.error("Bad tool group: %s", group)
-                    return 1
-
-                tool_info[group] = {}
-                for path in tg_dir.iterdir():
-                    # skip files like __label__ and __trigger__
-                    if not path.is_dir():
-                        continue
-
-                    host = path.name
-                    tool_info[group][host] = {}
-                    tool_info[group][host]["label"] = None
-                    tool_info[group][host]["tools"] = {}
-
+                toolsdict = tool_info[group][host]["tools"] = {}
+                if toolname:
                     # Check if the tool is in any of the hosts.
-                    if tool in self.tools(path):
-                        v = sorted((path / tool).read_text().rstrip("\n").split("\n"))
-                        tool_info[group][host]["tools"][tool] = v if opts else ""
+                    if toolname in self.tools(path):
+                        v = (path / toolname).read_text().rstrip("\n").split("\n")
+                        toolsdict[toolname] = sorted(v) if opts else ""
                         found = True
+                else:
+                    # no tool name was specified
+                    for tool in sorted(self.tools(path)):
+                        v = (path / tool).read_text().rstrip("\n").split("\n")
+                        toolsdict[tool] = sorted(v) if opts else ""
 
+        if toolname:
             if found:
                 self.print_results(tool_info, self.context.with_option)
                 return 0
             else:
-                msg = f'Tool "{self.context.name}" not found in '
+                msg = f'Tool "{toolname}" not found in '
                 msg += self.context.group[0] if self.context.group else "any group"
                 self.logger.error(msg)
                 return 1
 
+        # no tool name was specified
+        if tool_info:
```

Comment:
Based on the code comment, should this be in an `else` for the `if` at line 100?

Reply:
I failed to notice that the previous `if` results, one way or the other, in a `return`, so, _no_ this doesn't need to be in an `else`...except for the fact that Dave and I each independently requested it...so, I guess it's for the best.  😁

Reply:
Well, if you didn't notice it, then the readability of the code left something to be desired, so yeah, I guess it *is* for the best.